packages/kit/src/runtime/client/client.js

```diff
@@ -708,9 +708,7 @@ export function create_client({ target, base }) {
 		/** @type {import('types').ServerData | null} */
 		let server_data = null;
 
-		const url_changed = current.url
-			? url.pathname + url.search !== current.url.pathname + current.url.search
-			: false;
+		const url_changed = current.url ? id !== current.url.pathname + current.url.search : false;
```

Comment:
Are we sure the `id`, which serves as an unique identifier for caching from what I see, will always be an URL-type thing?

Reply:
Yeah, I know it is currently pathname + search - I'm just considering:
* `id` name is closer to "route id" than to `url`, which it currently represents in essence - I think that's what caused the initial oversight
* we are not sure if down the line that `id` will change to something else more specific, breaking this comparison once again.
So I opted to shed that `id` dependency - especially because `url` is still there being returned for its intended purpose.

Reply:
I agree that `id` is probably a misleading name in this instance because of `route.id`, but that `id` cannot change during the cause of a route change (only if another route change is triggered as a result of it, at which point a new `id` will be computed). I think we can keep it as is.

Reply:
Oh, I meant "change" as in "someone changes what is computed inside `get_navigation_intent` at some point due to a caching requirement change and overlooks this comparison due to destructuring, commits and it's borken again".

Thanks for merging!